### ISSUE TITLE
Multiplayer Rooms

### ISSUE THREAD

**Issue by LightbreakerQ** [Anuken/Mindustry-Suggestions#2357]
Ability to create unique public multiplayer rooms with an optional password (And the creator would have OP of course).

This will let you play with other people in a unique room when they are far.

The current way is to portforward but thats something very complicated for most people and people are not ok with sharing their IP.

Some servers are also an option but they don't serve everytime (especially because of the version differences) and also you can't customize the map, rules, mods etc.

In conclusion, we cant just play in a unique room with other people the way we want.

[X] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.

[X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.

[X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.

[X] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by Legomonster33** [comment]
thats what plugins are for

**Comment by Legomonster33** [comment]
passwords would be nice, steam already has public rooms

**Comment by LightbreakerQ** [comment]
> passwords would be nice, steam already has public rooms

What about mobile players?

**Comment by Kyllingene** [comment]
> Ability to create unique public multiplayer rooms with an optional password

This would require Anuke to have servers for you to use, which is not happening (this guy is a nearly unpaid, solo dev). Sorry, however you could certainly find a service which would let you host a server.

**Comment by LightbreakerQ** [comment]
> > Ability to create unique public multiplayer rooms with an optional password
> 
> This would require Anuke to have servers for you to use, which is not happening (this guy is a nearly unpaid, solo dev). Sorry, however you could certainly find a service which would let you host a server.
> 
> This goes under the "requires money" category of "not happening".

How about you instead support my suggestion so maybe Anuke would find a way? With great support this definitely can happen.

**Comment by Kyllingene** [comment]
> How about you instead support my suggestion so maybe Anuke would find a way?

If your suggestion is impossible, I will not support it. The only way Anuke could do this is if he got a lot of money, and relying on frequent, rather large donations for servers is not a good idea. Unless you happen to have a lot of influence in some server hosting platform and can get them to give Anuke some free servers, it can't happen. Dreaming is often good, but being unreasonable is not.

**Comment by LightbreakerQ** [comment]
> > How about you instead support my suggestion so maybe Anuke would find a way?
> 
> If your suggestion is impossible, I will not support it. The only way Anuke could do this is if he got a lot of money, and relying on frequent, rather large donations for servers is not a good idea. Unless you happen to have a lot of influence in some server hosting platform and can get them to give Anuke some free servers, it can't happen. Dreaming is often good, but being unreasonable is not.

its not impossible when its supported alot

**Comment by Kyllingene** [comment]
> relying on frequent, rather large donations for servers is not a good idea

**Comment by LightbreakerQ** [comment]
> > > Ability to create unique public multiplayer rooms with an optional password
> > 
> > 
> > This would require Anuke to have servers for you > How about you instead support my suggestion so 



> Simply stating support will not help, and
> 
> > relying on frequent, rather large donations for servers is not a good idea

There are dead games that dont make money at all but they still have running servers. There must be a way. Those games are dead but they still have a lot of running servers (The servers are ran by the developers).

**Comment by Ruichka** [comment]
If you *really* want to have such servers, please actually support the developer and buy Steam version. It has the feature you're asking for.

Please close this suggestion because #2094

**Comment by LightbreakerQ** [comment]
> Please close this suggestion because #2094

no lol

**Comment by Kyllingene** [comment]
> There are dead games that dont make money at all but they still have running servers

Those games are from *paid developers*. Again. Let me repeat. ***Paid.*** Anuke developer income consists of the mobile and Steam sales profits and a few donations here and there. You need to realize that Anuke cannot depend on donations for actual servers, especially ones that players use as they wish, as those cost a lot. Maybe, **maybe** if Anuke charged money for servers, then he could pay for them, but even that has problems and is unlikely to ever happen.

**Comment by LightbreakerQ** [comment]
> > There are dead games that dont make money at all but they still have running servers
> 
> Those games are from _paid developers_. Again. Let me repeat. _**Paid.**_ Anuke developer income consists of the mobile and Steam sales profits and a few donations here and there. You need to realize that Anuke cannot depend on donations for actual servers, especially ones that players use as they wish, as those cost a lot. Maybe, **maybe** if Anuke charged money for servers, then he could pay for them, but even that has problems and is unlikely to ever happen.

Mobile is free he doesnt get money from the mobile at all.

Also then this game needs some paid stuff because without it, it wont reach any point (probably the reason its unpopular)

**Comment by Kyllingene** [comment]
> Also then this game needs some paid stuff because without it, it wont reach any point (probably the reason its unpopular)

Excuse me, you are *not* in charge of what Anuke decides to do with his game. Also, if this game is unpopular, explain why the subreddit has 15k members.

This entire issue just shows that you are one of those people who tells a developer what to do without regard to the developer or the game. You simply want more free stuff. You need to realize that you are not the solo dev who gets $5 per purchase on the only platform he makes money on.

**Comment by LightbreakerQ** [comment]
> > Also then this game needs some paid stuff because without it, it wont reach any point (probably the reason its unpopular)
> 
> Excuse me, you are _not_ in charge of what Anuke decides to do with his game. Also, if this game is unpopular, explain why the subreddit has 15k members.
> 
> This entire issue just shows that you are one of those people who tells a developer what to do without regard to the developer or the game. You simply want more free stuff. You need to realize that you are not the solo dev who gets $5 per purchase on the only platform he makes money on.

15k isnt much

**Comment by Kyllingene** [comment]
And why do you get to decide how much is enough players? 15k is plenty for a healthy community, so why does he need more? So he gets more people like you demanding unreasonable things of an optionally free game?

**Comment by LightbreakerQ** [comment]
> And why do you get to decide how much is enough players? 15k is plenty for a healthy community, so why does he need more? So he gets more people like you demanding unreasonable things of an optionally free game?

you already decied how much is enough by saying

> if this game is unpopular, explain why the subreddit has 15k members.
> 15k is plenty for a healthy community

(the community isnt healthy tbh)

Also some members probably dont play it anymore.

And my suggestion wasnt unreasonable

**Comment by Kyllingene** [comment]
> (the community isnt healthy tbh)

Even if some people are toxic, making the community larger and more unmanageable is not going to help. Overwatch, Valorant, and so many other games have proven this.

> Also some members probably dont play it anymore.

Your point? Even 10k is perfectly fine.

> And my suggestion wasnt unreasonable

You are asking a solo, nearly unpaid dev to give you free, expensive servers so you don't have to do the tiny amount of work it takes to host your own.

**Comment by itcannotbe** [comment]
https://github.com/Anuken/Mindustry-Suggestions/issues/1153#issuecomment-725813205

**Comment by LightbreakerQ** [comment]
> Even if some people are toxic, making the community larger and more unmanageable is not going to help. Overwatch, Valorant, and so many other games have proven this.

Say 1/7 (instance) of players are toxic. Having more people means more toxic people but way more non-toxic people. It's also about the game and mindustry is a game that i dont think much toxic people would play either. (For instance if 1/2 of the players would be toxic in overwatch, then for mindustry it'd be 1/7). And last, overwatch and valorant are popular worldwide meaning all ages of people play it.

> Your point? Even 10k is perfectly fine.

What no way

> You are asking a solo, nearly unpaid dev to give you free, expensive servers so you don't have to do the tiny amount of work it takes to host your own.

As i said this game should make money for Anuke to develop it better. Also maybe in some way it can be possible without so much money. Just requires a research.

In conclusion, this game needs advertisement for popularity and money (through paid stuff) for development.

The game isn't bad but its horribly underrated because of the unpopularity. It's so much below average. It should have more popularity (about average) but not like overwatch, valorant (it wouldnt get to that point anyway).

And for the other part, money. The game should have paid stuff for money which is for development because with money anuke maybe can cooperate and make this game way better with a bigger development team, more servers etc.

(By paid stuff i mean paid game, in-app purchases...)

**Comment by Kyllingene** [comment]
> And last, overwatch and valorant are popular worldwide meaning all ages of people play it.

But this is exactly what would happen to Mindustry. You want Mindustry to become popular; it has pretty good multi-lingual support; nothing really keeping it from being cross-country.

> Say 1/7 (instance) of players are toxic. Having more people means more toxic people but way more non-toxic people.

If I were to start playing, and I met 20 players, I would have a high chance of meeting the same amount of toxic people and non-toxic people.

> What no way

***Explain why you need more than 10k players for a playerbase to be good.***

> The game isn't bad but its horribly underrated because of the unpopularity

What? That doesn't make any sense. If anything, people nowadays *look* for niche games.

**Comment by LightbreakerQ** [comment]
> But this is exactly what would happen to Mindustry. You want Mindustry to become popular; it has pretty good multi-lingual support; nothing really keeping it from being cross-country.

Look mindustry is a game that isn't popular at all. But it should be. Currently the game is getting more popular but at an extremely slow rate. Ads are the only way to fix this. There are other ways too like telling your friends about it but because of this game's complicated mechanics and stuff they probably wont like it (it was kind of a nerd game you know and with this logic stuff being official it totally is now).

> If I were to start playing, and I met 20 players, I would have a high chance of meeting the same amount of toxic people and non-toxic people.

How? I used to play this game long ago and didnt see much.

> _**Explain why you need more than 10k players**_

People tend to play more popular games. We need more people. If you want to play with your friends and make them enjoy it, this game has to be popular. You can still hardly make them play the game but they'll be like "oh this is so complicated", "boring game", "why do we even play this".

> What? That doesn't make any sense. If anything, people nowadays _look_ for niche games.

I disagree. People play popular games usually.

**Comment by LightbreakerQ** [comment]
> Anuke would have to hire and manage workers

Yup thats what we need.

> Saying things doesn't make them make sense. What do you mean? Sure, this game could be more popular, but why would making a higher barrier of entry, namely a price, make it more popular?

Price is for money, which is for development. Not popularity.

> Again, this would ruin Mindustry for lots of people. Look at Minecraft: it used to be a small dev team committedly making an indie game better. Now, it's another Microsoft moneygrab, with stores and in-game money and a ruined community. Ever notice that Java Edition, which only gets the updates, no stores or microtransactions, has the most community?

Mobile was ruined unluckily. Microsoft could do better but they didnt. And for java, why do you think its popular? It was memorical. The game was dead before pewdiepie started playing it again for memories. Mindustry isnt a game like that. In fact you cant compare mindustry to minecraft they are way different.

**Comment by Kyllingene** [comment]
> How? I used to play this game long ago and didnt see much.

Sorry, worded that wrong. I meant that as long as the ratio stays the same, picking a random sample of the same size will statistically yield the same ratios, no matter the community size.

> Ads are the only way to fix this

What? Either you mean paying to get ads, which again goes against the Mindustry ideal of a community without all the money-grabbing junk that goes on in too many games, or you mean getting money from ads, which not only makes no sense in the context you showed but also would kill its popularity, even inside the community.

> People tend to play more popular games. We need more people. If you want to play with your friends and make them enjoy it, this game has to be popular. You can still hardly make them play the game but they'll be like "oh this is so complicated", "boring game", "why do we even play this".

1. [Appeal to the people fallacy](https://en.wikipedia.org/wiki/Argumentum_ad_populum)
2. Different people like different games, so popularity means very little
3. If your friends don't like the game once they play it, how on this green earth would more popularity fix it?
4. ***Your friends opinions of the game aren't more important than the existing communities opinions, nor the devs.***

Stop trying to shoehorn in bad arguments for bad suggestions that would make the game worse in every conceivable way. Stop pretending that you are more important than the developer. Stop pretending that money makes things better. Stop pretending that math can say whatever you want it to say. Stop pretending that you know more than everyone else.

r/choosingbeggars r/iamverysmart

**Comment by Kyllingene** [comment]
> Yup thats what we need.

Actually, *WRONG*. Not only would that kill the player-made environment, it would drain Anuke. Do you think he wants to be an employer?

> Price is for money, which is for development. Not popularity.

**Comment by LightbreakerQ** [comment]
HOW would ads and a better development team negatively affect the existing players

**Comment by Kyllingene** [comment]
> HOW would ads and a better development team negatively affect the existing players

Because it would be a more closed-in, segregated dev team with less contact and *way* less influence from the players. As it stands, players can actually write code for the game and change it directly. Add a price tag, and Mindustry is suddenly no longer open-source (you can't have a paid, open-source game). The players are shut out. It is no longer Mindustry, but just another game-gone-wrong. Too many good dev teams go this path, and they all fail miserably. They might make money, but they lose the essence of what made their games unique and fun. Imagine if Toby Fox suddenly went commercial by this same logic. He would *fail*.

**Comment by LightbreakerQ** [comment]
> Sorry, worded that wrong. I meant that as long as the ratio stays the same, picking a random sample of the same size will statistically yield the same ratios, no matter the community size.

Thats why im saying more popularity doesnt affect the playersbases toxicness ratio.

> What? Either you mean paying to get ads, which again goes against the Mindustry ideal of a community without all the money-grabbing junk that goes on in too many games, or you mean getting money from ads, which not only makes no sense in the context you showed but also would kill its popularity, even inside the community.

No way ads would make the game less popular unless they are those "fake mobile game ads" type which i didnt suggest it at all

> 2. Different people like different games, so popularity means very little

Explain why games like factorio or that 3D factory building game is more pooular than mindustry. They are almost the same.

> 3. If your friends don't like the game once they play it, how on this green earth would more popularity fix it?

Psychology

> 4. _**Your friends opinions of the game aren't more important than the existing communities opinions, nor the devs.**_

First, by my friends i meant random people. An average, typical person. If they dont like it at all, then this game is not so good. You see this existing community is very small because most people dont like it at all.

> Stop trying to shoehorn in bad arguments for bad suggestions that would make the game worse in every conceivable way.

I am here trying to make this game better with good suggestions. You are just coming up with random bs. I asked for a single, multiplayer rooms feature. And look where you took it to.

> Stop pretending that you are more important than the developer.

Im not. Suggesting very important stuff for the game feature doesnt mean im more important than the devloper.

> Stop pretending that money makes things better. 

It does in this case. We cant have multiplayer rooms because we dont have money. We cant advertise because we dont have money. Without being able to advertise, the game becomes unpopular and you wouldnt make much money from it even if you make paid stuff. Anuke needs money to advertise it to make the game more popular. Then make paid stuff to get his money back, maybe even more. (With the game being unpopular he wouldn't get much).

> Stop pretending that math can say whatever you want it to say. 

When did i?

> Stop pretending that you know more than everyone else.

I didnt lol i just suggested important stuff i saw that wasnt going to happen. I thought that Anuke thought of these and felt like reminding him how important they are.

> r/choosingbeggars r/iamverysmart

Shut up silly redditor. You know what? Post it so maybe this game gets a bit more popularity.

**Comment by LightbreakerQ** [comment]
> Because it would be a more closed-in, segregated dev team with less contact and _way_ less influence from the players. As it stands, players can actually write code for the game and change it directly. Add a price tag, and Mindustry is suddenly no longer open-source (you can't have a paid, open-source game). The players are shut out. It is no longer Mindustry, but just another game-gone-wrong. Too many good dev teams go this path, and they all fail miserably. They might make money, but they lose the essence of what made their games unique and fun. Imagine if Toby Fox suddenly went commercial by this same logic. He would _fail_.

Then this game has to be non-open source. I didnt see much devs failing with this. Also for toby fox, his games were singleplayer.

**Comment by Kyllingene** [comment]
> Then this game has to be non-open source.

There is the critical weak point; at that, nobody but yourself will dare tread. Anuke is unlikely to ever take Mindustry out of that place, and that is a good thing.

> Shut up silly redditor. You know what? Post it so maybe this game gets a bit more popularity.

Guess what? I did. r/choosingbeggars likes it. I blurred out your name, of course; doxxing is objectively bad.

> Explain why games like factorio or that 3D factory building game is more pooular than mindustry. They are almost the same.

What are you trying to say? Anyway, they happen to be more popular because they have full dev teams that can afford ads and such. Mindustry is not one of those games.

> Psychology

If your friends are so easily swayed by the culture that they would start liking a game just because other people do, you need better friends.

> First, by my friends i meant random people. An average, typical person. If they dont like it at all, then this game is not so good. You see this existing community is very small because most people dont like it at all.

Actually, this is quite a good game; the community constantly thinks about the new players. And, would you look at that, *the community is the reason for that*. The fact that Mindustry has a diverse playerbase has resulted in a very welcoming game; one that is fun for illiterate children to hardcore extremist speedrunners. If the game isn't popular for that, why should we start trying? The game is good as it is, and pandering to the lowest common denominator is going to ostracize the community that made Mindustry what it is now.

> I didnt lol i just suggested important stuff i saw that wasnt going to happen.

You never once indicated you though this was not going to happen. Not once. And, throughout the entire discussion, you have repeatedly made it clear that you think you know better than the community and dev.

> When did i?

When you argued that more players means less toxicity. Let me jog your memory:

> Having more people means more toxic people but way more non-toxic people.

Finally, explain why this line:

> Shut up silly redditor.

Has any of the following:
1. Influence on the conversation
2. Common decency
3. A solid basis (you use reddit, don't lie, and it has done lots of good for many)
4. Adherence to the GitHub rules (this is actually reportable, although not much would happen)

Please, when in a discussion, use these two magical things: reason and discretion.

**Comment by Kyllingene** [comment]
Also, if this game stopped being open source, what would stop someone from republishing it as it was before? I could download the newest version, and then when it got closed up, republish it. The license allows for modification and distribution, so as long as I keep the same license and document my changes, I'm in the legal right. So, explain how making this closed-source makes sense in any degree?

**Comment by sample-text-here** [comment]
this conversation made me lose a lot of brain cells
1. mindustry is free and open source, so
2. anuke doesnt have money to host servers, but
3. the community *could* donate, but
4. there arent a lot of people who can donate, and
5. in order to keep the servers online, there would need to be constant donations, which means
6. there is a good change there will be constant server outages, and
7. port forwarding isnt complex and if you only share your ip with your friends its fine, plus
8. you could just pay for a server instead

with that said, lightbreakerq doesnt seem to understand basically anything so @Kyllingene its probably best to just ignore them

**Comment by IVV-NCX** [comment]
Well, I know a very simple port forwarding method. 

Step One: Download Zero Tier One.
Step Two: Create a private network.
Step Three: Give the network ID to a friend, approve their IP.
Step Four: Host a game.
Step Five: Friend inputs your IP as the name of the server.
Step Six: Done.

Although the IP is public, it's not your exact IP. Don't ask more questions, except regarding how to use this method.

**Comment by Kyllingene** [comment]
Ngrok is also helpful. It is basically a port-forward, except you use one of the servers IP's, and it works on a session-to-session basis (the IP you connect to changes each session).

**Comment by SheepCommander** [comment]
I read through some of this, was about to scream (It's 2 in the morning) and then realized I'm less than a quarter of the way through scrolling.

WHAT THE ACTUAL ADEFOI;JWR PERSON CLOSE THE DAMNED SUGGESTION
(Also would like to point out as a side-rant:
> This game needs to be un-open sourced! This game needs to be paid! That's how you be successful!

> Then buy the steam version it has the multiplayer [public and private] that you want

> no lol

can't wait for covid to evolve at the speed of SCP-682 and kill us all

**Comment by LightbreakerQ** [comment]
> Guess what? I did. r/choosingbeggars likes it.

link?

**Comment by itcannotbe** [comment]
comments/mxsj0t/

**Comment by SheepCommander** [comment]
lmao nice

**Comment by marky0470** [comment]
found the microsoft employee

**Comment by Kyllingene** [comment]
@LightbreakerQ, please close this. It has been made clear by the community that this will not be happening. All it is right now is an issue for bored surfers of the web to look at and go "huh, poor guy". This is just another issue for Anuke to filter through. Please close.

**Comment by SheepCommander** [comment]
> Opened this issue (their first ever)
> Joined GitHub this month
If you're not a bot close it
